Bump DScanner & enable for more unused variable checks

MoonlightSentinel opened the pull request:
This removes the following modules from the blacklist:

- dmd.dinterpret
- dmd.dmangle
- dmd.e2ir
- dmd.libmscoff
- dmd.ob
- dmd.semantic2
- dmd.tocvdebug

Geod24 commented on src/dmd/ob.d:
```diff
@@ -1676,7 +1676,7 @@ void genKill(ref ObState obstate, ObNode* ob)
             override void visit(NewExp e)
             {
-                Type tb = e.newtype.toBasetype();
+                // Type tb = e.newtype.toBasetype();
```
No need to comment things out, just remove them.

MoonlightSentinel replied:
Done

MoonlightSentinel committed:
Bump DScanner to v0.9.0-beta.1

MoonlightSentinel committed:
Enforce unused variable check for more modules

This removes the following modules from the blacklist:

- dmd.dinterpret
- dmd.dmangle
- dmd.e2ir
- dmd.libmscoff
- dmd.ob
- dmd.semantic2
- dmd.tocvdebug

MoonlightSentinel commented on src/dmd/ob.d:
```diff
@@ -1514,7 +1511,7 @@ void genKill(ref ObState obstate, ObNode* ob)
                 {
                     const vi = obstate.varStack[i];
-                    auto pvs = &ob.gen[vi];
+                    // auto pvs = &ob.gen[vi];
```
Keeping this because it's "used" below

thewilsonator commented on src/build.d:
```diff
@@ -525,21 +525,13 @@ alias style = makeRule!((builder, rule)
-            run([git, "-C", dscannerDir.buildPath("libdparse"), "checkout", "63559db5cc6fa38c01bdda36e09638b5f20fb8e5"]);
-        })
+        .command([
+            // FIXME: Omitted --shallow-submodules because it requires a more recent
+            //        git version which is not available on buildkite
```
This comment seems out of date.

MoonlightSentinel replied:
Last run from BuildKite:

```
[...]
Command: git clone --depth=1 --recurse-submodules --shallow-submodules --branch=v0.9.0-beta.1 https://github.com/dlang-community/Dscanner /var/lib/buildkite-agent/builds/ci-agent-1b8888d4-05cf-49df-8be7-5111d465b14e-1/dlang/dmd/dmd/generated/linux/release/64/dscanner
-------------------------------------------------------------
ERROR: error: unknown option `shallow-submodules'
usage: git clone [<options>] [--] <repo> [<dir>]
[...]
```

thewilsonator replied:
Ah its not passing.

MoonlightSentinel replied:
For reference, `--shallow-submodules` is available since `git v2.9.0` which was released in July 2016.
